### PR TITLE
Fix customer creation birthdate bug #253

### LINKED ISSUE #253: [Bug]: Customer creation broken
Something is wrong with the Birthday. It sends an empty object to the backend, which fails.

### PR DESCRIPTION
**Pull request by beesaferoot**
### Describe how you tested your changes?

This PR fixes #253  and also fixes payment history query bug on customer detail page.

### Pull Request checklist

Please confirm you have completed any of the necessary steps below.

- [ ] Meaningful Pull Request title and description
- [ ] Changes tested as described above
- [ ] Added appropriate documentation for the change.
- [ ] Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

**Review by dmohns — COMMENTED** [review]
Few comments

File: src/backend/app/Models/PaymentHistory.php [hunk 1]
```diff
@@ -34,9 +34,10 @@ public function transaction(): BelongsTo {
     }
 
     public function getFlow(string $payer_type, int $payer_id, string $period, $limit = null, string $order = 'ASC') {
-        $sql = 'SELECT sum(amount) as amount, payment_type, CONCAT_WS("/", '.$period.') as aperiod from'.
-            ' payment_histories where payer_id=:payer_id and payer_type=:payer_type '.
-            'GROUP by concat( '.$period.'), payment_type ORDER BY created_at  '.$order;
+        $sql = 'SELECT sum(amount) as amount, payment_type, CONCAT_WS("/", '.$period.') as aperiod '.
+            'FROM payment_histories '.
+            'WHERE payer_id=:payer_id AND payer_type=:payer_type '.
+            'GROUP by aperiod, payment_type ORDER BY aperiod  '.$order;
```
[the remark this answers is not in the document]
Comment (beesaferoot) [reply src/backend/app/Models/PaymentHistory.php:40]: Created an issue for this #448

File: src/backend/app/Services/PersonService.php [hunk 2]
```diff
@@ -112,12 +112,14 @@ public function isMaintenancePerson($customerType): bool {
     }
 
     public function createPersonDataFromRequest(Request $request): array {
+        $birthDate = Carbon::parse($request->get('birth_date'))->format('Y-m-d H:i:s');
```
[the remark this answers is not in the document]
Comment (beesaferoot) [reply src/backend/app/Services/PersonService.php:115]: I could try it out. Thanks

**Comment by beesaferoot** [comment]
Ready for review.

**Review by munyanezaarmel — APPROVED** [review]

**Pull request merged**